### DISCUSSION TITLE
Icicle Dev Kit not booting into Linux successfully from SD card, and also unable to see eMMC as a mounted device on either Linux or Windows

### DISCUSSION THREAD

**Discussion by es-ggg** [polarfire-soc/polarfire-soc-documentation#374] [HSS and U-Boot]
Hello,

I'm new to using PolarFire dev kit and have issues setting up.

After successfully flashing an SD card with either a yocto BSP image I built using the command: MACHINE=icicle-kit-es bitbake mpfs-dev-cli or the one that was provided in the releases here: https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/releases. The polarfire icicle development kit fails to boot successfully with the SD card inserted. It stops at this line “Starting kernel …” in the serial UART1 terminal. Please see the attached logs. It fails to enter the Linux Kernel portion as it attempts booting.
[Logs of U-Boot Failure.txt](https://github.com/polarfire-soc/polarfire-soc-documentation/files/13320878/Logs.of.U-Boot.Failure.txt)

FYI, when I attempted flashing the SD card, the command bmaptool copy tmp-glibc/deploy/images/icicle-kit-es/mpfs-dev-cli-icicle-kit-es.wic /dev/sdX failed on my Linux machine by indicating the block device could not be opened. I decided to flash the image this way: zcat  tmp-glibc/deploy/images/icicle-kit-es/mpfs-dev-cli-icicle-kit-es.wic.gz | sudo dd of=/dev/sdX bs=4096 iflag=fullblock oflag=direct conv=fsync status=progress. It was time consuming, but it successfully wrote to the SD card and I corrected the secondary GUID Partition table by doing this: fdisk /dev/sdX and entering w. It had the boot partition and the root partition on the SD card. I observed that both the boot.scr.uImg file and fiImage were dated April 5th 2011 regardless of the image I wrote to the card i.e. the one I built had that date and the one from the releases website above also had that date.

I am attempting to boot from SD card because I'm new to this project and I'm yet to successfully boot an image that I've written to an SD card, so I want to avoid overwriting the default good image on the eMMC which boots successfully. Another thing I observed when I attempted creating a backup image of the eMMC was that the device fails to show up on my PC as a mounted device, hence I could not use the win32diskimager app to create a backup. 

Is there a required driver to be installed to make the eMMC mount on my computer? If so, where do I find it?

Also between booting from eMMC and SD card, is there a toggle (e.g. button/software/devicetree modification, etc.) necessary to indicate to the dev kit where it's booting from? I observed, that it attempts to boot from SD card when there's an SD card inserted, and boots from eMMC when there's no SD card inserted.

Thanks in advance.

**Comment by rockoguido** [comment]
Hi.   a few things to keep in mind.   One is that what is in the fabric has to align with the version of Yocto Linux you are attempting to boot.   It appears given your error message that you are not aligned.   Easy problem to fix is the good news.   Will just require a few steps..

A new kit has Linux that boots via the eMMC, has that worked for you?   So remove the SD card, check the jumpers in the Getting Started Guide I list below and see if you come up, good initial test
see the following.. for getting started in general
https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/reference-designs-fpga-and-development-kits/icicle-kit-user-guide.md
Also see https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/releases    for a pre-built core-image-minimal image that you can deploy via the SD card and see if that comes up

if you need to start at square 1 go to 
https://github.com/polarfire-soc/icicle-kit-reference-design/releases/
and deploy the Icicle Kit Base Design, you'll need to use FPExpress to update the Bitstream
Then you can use the pre-built Linux image I mentioned above and work on getting that to boot.

Once environment has been validated then you can use your Yocto build and deploy

Hope that helps and feel free to ask any follow-on questions

**Comment by es-ggg** [comment]
Thank you so much for your quick response. I'll try this tomorrow and get back to you if it works.

I already had some success by flashing the exact version of the image on the eMMC that's located on the releases website here: https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/releases on the SD card, and the SD card boots successfully. So before updating the HSS, I'll checkout that same version on the BSP polarfire icicle repo https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp to see if it works. Once that works, I'll move to updating the fabric the way you described above with the FPExpress for the Bitstream and so on.

Thank you very much for the help!!!

**Comment by es-ggg** [comment]
FYI, After updating the BSP on the eMMC and/or SD card to the latest version here: https://github.com/polarfire-soc/icicle-kit-reference-design/releases, i.e. v2023.09, at the time this was being written; and then updating the bitstream on the icicle dev kit with the latest release from here: https://github.com/polarfire-soc/icicle-kit-reference-design/releases - i.e. v2023.06 at the time this was written. The icicle dev kit failed to boot. It got past the "Starting kernel" mentioned in the original question, but there was a kernel panic error that prevented the boot cycle from completing. 

I got around this issue, by using the previous BSP version - v2023.06 on both the SD card and eMMC to pair with the bitstream of v2023.06 from the respective URLs mentioned previously. I believe there's a compatibility issue between the current latest releases of the BSP and the bitstream. Could this be looked into?

Thanks

**Reply by rockoguido** [reply]
hello, yes thanks for pursuing this,   in general a really good idea to keep the bitstream and the Linux distro in sync, in your case 2023.06.    This issue has been seen by others on their proprietary platforms that are PolarFire SoC based.     We will look into this per the Icicle Kit, I know personally I'm able to run the latest pre-built core-image-minimal image (2023.09) on top of the latest bitstream. along with a Yocto version I built via 2023.09 too.
In the meantime hope you are ok staying with 2023.06 and of course when you build you own via Yocto stay with the 2023.06 branch, so not the main branch.
Regards,
Dave
